dts/arm/nordic/nrf51822.dtsi

```diff
@@ -78,6 +78,7 @@
 			reg = <0x40003000 0x1000>;
 			interrupts = <3 NRF_DEFAULT_IRQ_PRIORITY>;
 			max-frequency = <DT_FREQ_M(8)>;
+			easydma-maxcnt-bits = <8>;
```

Comment:
If you wish to have it here, please consider taking it from HAL. DTS is processed via C preprocessor and you can include a file with definitions.

Reply:
Devicetree is the source of hardware description, not HALs. See for example `max-frequency` above. The fact we've been using HALs to retrieve stuff like IRQ numbers, addressed, etc. is a mistake that needs to be fixed in the long term. Using named defines, ie `SPIM0`, `SPIM1` etc. is also problematic: we can't use the standard instantiation mechanism in Zephyr. This needs to be removed sooner than later, to align with all other vendors in-tree.

Reply:
I believe we have some instance agnostic definitions:
SPIM_RXD_MAXCNT_MAXCNT_Msk, SPIM_RXD_AMOUNT_AMOUNT_Msk, SPIM_RXD_LIST_LIST_Msk... and the same for TXD side. Maybe one of them would make the job?

Reply:
> I believe we have some instance agnostic definitions: SPIM_RXD_MAXCNT_MAXCNT_Msk, SPIM_RXD_AMOUNT_AMOUNT_Msk, SPIM_RXD_LIST_LIST_Msk... and the same for TXD side. Maybe one of them would make the job?

If they exist, then it's also a good solution. In the end these options are instance agnostic by looking at all datasheets. It's just the bits in MAXCNT. This is why I was initially pushing for this.

Reply:
Opened https://github.com/zephyrproject-rtos/zephyr/pull/54672